optimizely/optimizely.py

```diff
@@ -63,20 +63,22 @@ def __init__(self,
 
     try:
       self.config = project_config.ProjectConfig(datafile, self.logger, self.error_handler)
-    except:
+
+    except Exception as error:
       self.is_valid = False
-      self.config = None
       # We actually want to log this error to stderr, so make sure the logger
-      #   has a handler capable of doing that.
+      # has a handler capable of doing that.
       self.logger = _logging.reset_logger(self.logger_name)
-      self.logger.error(enums.Errors.INVALID_INPUT_ERROR.format('datafile'))
-      return
 
-    if not self.config.was_parsing_successful():
-      self.is_valid = False
-      # We actually want to log this error to stderr, so make sure the logger
-      #   has a handler capable of doing that.
-      self.logger.error(enums.Errors.UNSUPPORTED_DATAFILE_VERSION)
+      if type(error) is exceptions.UnsupportedDatafileVersionException:
```

Comment:
I'd recommend instead of doing this having multiple `except` clauses

Reply:
Hmm, I actually asked them to combine the multiple `except` clauses because the only thing that differs is the messaging. Otherwise we'd be duplicating (just 1 thing right now) the part where we pass the error to the handler.

Reply:
After thinking about it a bit more, I think having the multiple clauses is the more idiomatic way of dealing with this. We can instead, save the message and error objects in variables in each of the `except` clauses and introduce a `finally` clause that checks if we recorded any errors, and if so, then log + send to error handler. WDYT? @aliabbasrizvi

Reply:
Hey @mikeng13 ... yes that is more idiomatic and I believe we should adopt that.